src/dbt_client/dbtCloudIntegration.ts

```diff
@@ -713,6 +714,54 @@ export class DBTCloudProjectIntegration
     );
   }
 
+  async getBulkCompiledSQL(models: NodeMetaData[]) {
+    const queries: string[] = [];
+    for (const node of models) {
+      if (!node.path) {
+        continue;
+      }
+      const fileContentBytes = await workspace.fs.readFile(Uri.file(node.path));
+      const query = fileContentBytes.toString();
+      const tag = `<--- ${node.uniqueId} --->`;
+      queries.push(`${tag} ${query} ${tag}`);
+    }
+    const inlineQuery = queries.join(";");
+    console.log(inlineQuery);
+    const compileQueryCommand = this.dbtCloudCommand(
+      new DBTCommand("Getting catalog...", [
+        "compile",
+        "--inline",
+        inlineQuery.trim().split("\n").join(" "),
```

Comment:
This will give different results from before as we are not compiling nodes but just inline queries. Also by mixing the queries, state and variables might give unexpected results.

Reply:
yeah. also its failing for ephemeral cases.